### ISSUE TITLE
zc157: currency_cron.php throws PHP warning on PHP 7.2+

### ISSUE THREAD

**Issue by lat9** [zencart/zencart#3464]
The log, similar to:
```
[15-May-2020 15:11:27 Europe/Berlin] Request URI: cron, IP address: cron
#1  session_set_save_handler() called at [C:\xampp\htdocs\zc157a\includes\functions\sessions.php:102]
#2  require_once(C:\xampp\htdocs\zc157a\includes\functions\sessions.php) called at [C:\xampp\htdocs\zc157a\includes\autoload_func.php:27]
#3  require(C:\xampp\htdocs\zc157a\includes\autoload_func.php) called at [C:\xampp\htdocs\zc157a\nortH-tWN-braiN\includes\application_top.php:36]
#4  require(C:\xampp\htdocs\zc157a\nortH-tWN-braiN\includes\application_top.php) called at [Standard input code:56]
--> PHP Warning: session_set_save_handler(): Cannot change save handler when headers already sent in C:\xampp\htdocs\zc157a\includes\functions\sessions.php on line 102.
```

is thrown, I believe, due to the shebang (`#!/usr/bin/php`) on the first line and the fact that a session, for whatever reason, is being established during this cron.

Why is a cron-job creating a session in the first place?

**Comment by drbyte** [comment]
Interesting. I don't see such errors. This is on PHP 7.4:

```
> cd /dev-1-5-7
> php admin/currency_cron.php

Notice: Constant TEXT_PRODUCT_WEIGHT_UNIT already defined in /dev-1-5-7/admin/includes/languages/english.php on line 563
Updating currencies...
The exchange rate for US Dollar (USD) was updated successfully to 1 via ecb.
The exchange rate for Euro (EUR) was updated successfully to 0.9724023 via ecb.
The exchange rate for GB Pound (GBP) was updated successfully to 0.86289035 via ecb.
The exchange rate for Canadian Dollar (CAD) was updated successfully to 1.48106594 via ecb.
The exchange rate for Australian Dollar (AUD) was updated successfully to 1.63412206 via ecb.
Done.
```

**Comment by lat9** [comment]
OK, perhaps it's my running that in a Windows `cmd` window.  Still, why is a cron-job creating a session?

**Comment by drbyte** [comment]
> Still, why is a cron-job creating a session?

Multi-language support, among other things.

**Comment by drbyte** [comment]
> I believe, due to the shebang (`#!/usr/bin/php`) on the first line

Perhaps; however, the `#` should cause the line to be ignored. 
You're right: it may be a Windows limitation.

**Comment by drbyte** [comment]
Since the shebang's purpose is to (where possible) allow one to call the script directly on the command line without having to specify the php executable, you could remove it in your production environment and just be sure to always invoke php first and pass the script to it.

tldr: the shebang is optional: feel free to remove it if your environment can't support it

**Comment by lat9** [comment]
OK, I'm good if you're good.
